clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -93,25 +93,36 @@
     private final String metricGrpPrefix;
     private final Map<String, String> metricTags;
     private final ChannelBuilder channelBuilder;
-    private final Map<String, Long> lruConnections;
-    private final long connectionsMaxIdleNanos;
     private final int maxReceiveSize;
     private final boolean metricsPerConnection;
-    private long currentTimeNanos;
-    private long nextIdleCloseCheckTime;
-
+    private final IdleExpiryManager idleExpiryManager;
 
     /**
      * Create a new nioSelector
+     *
+     * @param maxReceiveSize Max size in bytes of a single network receive
```

Comment:
By the way, do you think unlimited is ever a good idea? This is why we get those OOM errors when we attempt to allocate memory based on garbage data from an invalid response.